scripts/tank_cmd.py

```diff
@@ -1230,7 +1254,148 @@ def run_engine_cmd(log, pipeline_config_root, context_items, command, using_cwd,
         return tank_command.run_action(log, tk, ctx, command, args)
 
 
+def _extract_args(cmd_line, *args):
+    """
+    Extract specified arguments and sorts them. Also removes them from the
+    command line.
+
+    :param cmd_line: Array of command line arguments.
+    :param args: Argument list to extract
+
+    :returns: A list of key, value tuples representing each arguments.
+    """
+    # Find all instances of each argument in the command line
+    found_args = []
+    for cmd_line_token in cmd_line:
+        for arg in args:
+            if cmd_line_token.startswith("--%s=" % arg):
+                found_args.append(cmd_line_token)
+
+    # Strip all these arguments from the command line.
+    for a in found_args:
+        cmd_line.remove(a)
+
+    credentials = []
+
+    # Create a list of tuples for each argument we found.
+    for arg in found_args:
+        pos = arg.find("=")
+        # Take everything after the -- up to but not including the =
+        arg_name = arg[2:pos]
+        arg_value = arg[pos + 1:]
+        credentials.append((arg_name, arg_value))
+
+    return credentials
+
+
+def _validate_only_once(args, arg):
+    """
+    Validates that an argument is only present once.
+
+    :param args: List of tuples of arguments that were extracted from the command line.
+    :param arg: Argument to validate
+
+    :raises InvalidCredentials: If an argument has been specified more than once,
+                                this exception is raised.
+    """
+    occurences = filter(lambda a: a[0] == arg, args)
+    if len(occurences) > 1:
+        raise InvalidCredentials("argument '%s' specified more than once." % arg)
+
+
+def _validate_args_cardinality(args, arg1, arg2):
+    """
+    Makes sure that there are no more and no less than 1 value for each argument.
+
+    :param args: List of argument tuples.
+    :param arg1: First argument to test for.
+    :param arg2: Second argument to test for.
+
+    :raises InvalidCredentials: When there is more or less than 1 value for
+                                a given argument, this exception is raised.
+    """
+    # Too few parameters, find out which one is missing, let the user know
+    # which is missing.
+    if len(args) < 2:
+        raise InvalidCredentials("missing argument '%s'" % (arg1 if args[0][0] == arg2 else arg2))
+
+    # Make sure each arguments are not specified more than once.
+    _validate_only_once(args, arg1)
+    _validate_only_once(args, arg2)
+
+
+def _read_credentials_from_file(auth_path):
+    """
+    Reads the credentials from a file and returns two tuples: one for
+    login/password values and the other for script/key values. Any other values
+    are ignored.
+
+    :param auth_path: Path to a file with credentials.
+
+    :returns: A tuple of list of key, value pairs for values parsed. For example,
```

Comment:
Giving people the option to store their password seems like really bad practice to me. I might be wrong here, but my gut feeling here is that we shouldn't allow that. The external file should only allow you to store script key and script name, *not* login/password. This feels like it would lead our users down the right path, forcing them to create shotgun scripts for services requiring this file based authentication. Can you see any use cases where we would need to store username/password in a file?

Remember, we can always add it later if we need it. But once it has been added, we cannot easily remove it.